### PR TITLE
feat: ignorePrAuthor

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Adds a new config option `ignorePrAuthor` that removes PR author filtering from platform `getPrList()`.

## Context:

This is needed if someone needs to migrate bot accounts, including from hosted app to self-hosted.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Ran on a real repository (GitLab-only)

### CODE REVIEW

**Comment by zharinov** [comment]
Not sure, but maybe something in `lib/util/git` may require changes. Or everything should be fine?

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -820,6 +820,12 @@ Renovate will try to configure this to `true` also if you have configured any `n
 Using this setting, you can selectively ignore package files that you don't want Renovate autodiscovering.
 For instance if your repository has an "examples" directory of many package.json files that you don't want to be kept up to date.
 
+## ignorePrAuthor
```
Comment (JamieMagee) [inline docs/usage/configuration-options.md:823]: Maybe add the example from the PR description as well (app to self-hosted).=

**Comment by JamieMagee** [comment]
No changes on Azure + Gitea. Are they lacking the functionality right now?

File: lib/platform/bitbucket-server/index.ts [hunk 2]
```diff
@@ -294,11 +296,14 @@ export async function getPrList(refreshCache?: boolean): Promise<Pr[]> {
   logger.debug(`getPrList()`);
   // istanbul ignore next
   if (!config.prList || refreshCache) {
-    const query = new URLSearchParams({
+    const searchParams = {
       state: 'ALL',
       'role.1': 'AUTHOR',
```
Comment (viceice) [inline lib/platform/bitbucket-server/index.ts:301]:
```suggestion
```

I think this needs to be removed too

File: lib/platform/bitbucket-server/index.ts [hunk 3]
```diff
@@ -294,11 +296,14 @@ export async function getPrList(refreshCache?: boolean): Promise<Pr[]> {
...
-      'username.1': config.username,
-    }).toString();
+    };
+    if (!config.ignorePrAuthor) {
+      searchParams['username.1'] = config.username;
```
Comment (viceice) [inline lib/platform/bitbucket-server/index.ts:304]:
```suggestion
      searchParams['username.1'] = config.username;
      searchParams['role.1'] = 'AUTHOR';
```
and added here

**Comment by rarkins** [comment]
> No changes on Azure + Gitea. Are they lacking the functionality right now?

Yeah, I couldn't see any place where they filter their PR list calls by author.

Azure: https://github.com/renovatebot/renovate/blob/e3ccc35d4a82feaa21c02cdedeaa6a3c4d2bce0e/lib/platform/azure/index.ts#L186-L193

Gitea: https://github.com/renovatebot/renovate/blob/e3ccc35d4a82feaa21c02cdedeaa6a3c4d2bce0e/lib/platform/gitea/index.ts#L408-L413

File: lib/platform/gitlab/index.ts [hunk 4]
```diff
@@ -370,10 +373,13 @@ function massagePr(prToModify: Pr): Pr {
...
 async function fetchPrList(): Promise<Pr[]> {
-  const query = new URLSearchParams({
+  const searchParams = {
     per_page: '100',
-    author_id: `${authorId}`,
-  }).toString();
+  } as any;
+  if (!config.ignorePrAuthor) {
+    searchParams.author_id = `${authorId}`;
+  }
```
Comment (viceice) [inline lib/platform/gitlab/index.ts:381]:
```suggestion
  if (config.ignorePrAuthor) {
    // https://docs.gitlab.com/ee/api/merge_requests.html#list-merge-requests
    // default: `scope=created_by_me`
    searchParams.scope = 'all';
  }
```

https://docs.gitlab.com/ee/api/merge_requests.html#list-merge-requests

**Review by viceice — CHANGES_REQUESTED** [review]
some small changes

@JamieMagee Yes, the other platforms don't have such a filter (or we maybe aren't aware of, like gitlab scope)

**Comment by JamieMagee** [comment]
Okay, I'll take a look at the current state of the Azure DevOps API for this feature.

**Comment by viceice** [comment]
what about issue filter?
https://github.com/renovatebot/renovate/blob/bffd90de6684eae8753ca234cb207139e4bd21f9/lib%2Fplatform%2Fgitlab%2Findex.ts#L684

**Review by viceice — APPROVED** [review]

**Pull request merged**